### DISCUSSION TITLE
Blank screen when ESP32 goes into deep sleep

### DISCUSSION THREAD

**Discussion by reneverschoor** [moononournation/Arduino_GFX#186] [General]
Thank you very much for this library.
I settled on Arduino_GFX from the start of my project, and have been very happy with it.

Right now I'm facing a problem that is not directly related to Arduino_GFX.
As you have a lot of experience driving displays, I'm asking it here hopig you might have some insight.

I have a LilyGO TTGO T8 ESP32 with SD card slot driving two Waveshare 1.8" 128X160 SPI TFT screens with ST7735 controller.
All works great, I am able to display two videos at 8 FPS.

As an extra I wanted to implement deep sleep.
The system doesn't have to run continuously, and sleeping would save the life span of the screens.
When a contact is touched, the whole system goes into deep sleep.
Touch the contact again, and the system starts up and the videos start showing again.
Implementing the touch, deep sleep, and waking up, proved to be no problem.

However, when the ESP32 goes to deep sleep, the screens turn white with full(?) background lightning.
Clearly that's not what I wanted to achieve, the screens are still fully on while the ESP32 is sleeping.

The screen's VCC is connected to the 3.3V pin of the TTGO.
I guess that when the ESP32 goes to sleep, all signal pins go floating or get pulled up/down, and with only VCC and GND applied, a white screen is the result.

Is there any way I can blank the screen when the ESP32 goes into deep sleep?

**Comment by moononournation** [comment]
first you need to turn off the LCD backlight, you may need some research how to keep the GPIO that control backlight in LOW status while deep sleep.

**Reply by reneverschoor** [reply]
Ah, perfect!
Thank you for putting me on the right track, got it all working now :)

**Reply by moononournation** [reply]
The next step is try to call displayOff() before deep sleep, then call displayOn() after wake up.

**Reply by reneverschoor** [reply]
Good call.
I've now added a `gfx->displayOff` call just before going into deep sleep.
As waking up from deep sleep is basically a hard reset, there's no need to do a separate `displayOn` call.
In `tftInit` you're already sending the needed `ST7735_SLPOUT` signal.

I'm happy :)
